packages/pieces/google-drive/src/lib/action/export-document-as-pdf.action.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+
+
+import { googleDriveAuth } from '../../index';
+import { Property, createAction } from "@activepieces/pieces-framework";
+import { google } from 'googleapis';
+import { OAuth2Client } from 'googleapis-common';
+import { Stream } from 'stream';
+
+export const exportDocumentAsPdf = createAction({
+    displayName: "Export Document as PDF",
+    auth: googleDriveAuth,
+    name: "export_document_as_pdf",
+    description: "Export a document as PDF",
+    props: {
+        documentId: Property.ShortText({
+            displayName: "Document ID",
+            description: "The ID of the document to export",
+            required: true,
+        }),
+        folderId: Property.ShortText({
+            displayName: "Folder ID",
+            description: "The ID of the folder where the file will be exported",
+            required: true,
+        }),
+        name: Property.ShortText({
+            displayName: "Name",
+            description: "The name of the new file (do not include the extension)",
+            required: true,
+        }),
+    },
+    async run(context) {
+        const authClient = new OAuth2Client();
+        authClient.setCredentials(context.auth)
+
+        const documentId = context.propsValue.documentId;
+        const folderId = context.propsValue.folderId;
+        const nameForNewFile = context.propsValue.name;
+
+        const drive = google.drive({ version: 'v3', auth: authClient });
+
+        const result = await drive.files.export({
+            fileId: documentId,
+            mimeType: 'application/pdf',
+
+        }, {responseType: "arraybuffer"});
+
+       const requestBody = {
+            name: nameForNewFile + ".pdf",
+            parents: [folderId],
+        };
+        const templateBuffer = Buffer.from(result.data as any, 'base64');
+
+        const streamm = new Stream.PassThrough().end(templateBuffer);
+
+        const media = {
+            mimeType: 'application/pdf',
+            body: streamm,
+        };
+
+        const file = await drive.files.create({
```

Comment:
The display name of the action is misleading, I think this one export as pdf and it creates new file inside the google drive right?

Also do you think this action should be in Google Docs rather drive?

Reply:
yeah, i fixed the name of the action... But actually, this is a Google drive thing. It uses the google drive API. (I tried to include it on the Docs Piece, but I need the Google drive auth)

https://developers.google.com/drive/api/reference/rest/v3/files/export